### PR TITLE
[NN] nn modules & examples update

### PR DESCRIPTION
**Pull request by yzh119**
## Description
This PR implements some nn modules in MXNet as we did in PyTorch.

Other updates:
- fixed the memory consumption problem of GGNN mentioned in #852 .
- add MXNet examples of appnp, graphsage, sgc, monet, gin.
- add PyTorch examples of chebnet & monet on MNIST.
- minor changes with APIs.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/nn/mxnet/conv/chebconv.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                 in_feats,
+                 out_feats,
+                 k,
+                 bias=True):
+        super(ChebConv, self).__init__()
+        self._in_feats = in_feats
+        self._out_feats = out_feats
+        self._k = k
+        with self.name_scope():
+            # NOTE(zihao): MXNet do not support ModuleList, use Sequential as workaround.
```
Comment (jermainewang) [inline python/dgl/nn/mxnet/conv/chebconv.py]: I think Sequential is the right way. Consider remove this note.

File: python/dgl/nn/pytorch/conv/gatedgraphconv.py [hunk 2]
```diff
@@ -75,13 +80,17 @@ def forward(self, graph, feat, etypes):
...
-            a = graph.ndata.pop('a').sum(dim=1) # (N, D)
+            graph.ndata['h'] = feat
+            for i in range(self._n_etypes):
+                eids = (etypes == i).nonzero().view(-1)
+                if len(eids) > 0:
+                    graph.apply_edges(
+                        lambda edges: {'W_e*h': self.linears[i](edges.src['h'])},
+                        eids
+                    )
+            graph.update_all(fn.copy_e('W_e*h', 'm'), fn.sum('m', 'a'))
```
Comment (jermainewang) [inline python/dgl/nn/pytorch/conv/gatedgraphconv.py]: Why is this a better implementation?

Comment (yzh119) [reply python/dgl/nn/pytorch/conv/gatedgraphconv.py]: In this way, we do not have to move store `W` on edges, thus reducing the GPU memory consumption, see #852 .

Comment (jermainewang) [reply python/dgl/nn/pytorch/conv/gatedgraphconv.py]: I see. From the math, it is quite similar to RGCN. Maybe we could try use heterograph APIs later.

**Pull request merged**